WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.kt

```diff
@@ -0,0 +1,353 @@
+    }
+
+    private var mOnNoteClickListener: OnNoteClickListener? = null
+    fun setFilter(newFilter: FILTERS) {
+        currentFilter = newFilter
+    }
+
+    fun addAll(notes: List<Note>, clearBeforeAdding: Boolean) {
+        notes.sortedWith(TimeStampComparator())
+        try {
+            if (clearBeforeAdding) {
+                mNotes.clear()
+            }
+            mNotes.addAll(notes)
+        } finally {
+            myNotifyDatasetChanged()
+        }
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private fun myNotifyDatasetChanged() {
+        buildFilteredNotesList(mFilteredNotes, mNotes, currentFilter)
+        notifyDataSetChanged()
+        mDataLoadedListener.onDataLoaded(itemCount)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.notifications_list_item, parent, false)
+        return NoteViewHolder(view)
+    }
+
+    private fun getNoteAtPosition(position: Int): Note? {
+        return if (isValidPosition(position)) {
+            mFilteredNotes[position]
+        } else null
+    }
+
+    private fun isValidPosition(position: Int): Boolean {
+        return position >= 0 && position < mFilteredNotes.size
+    }
+
+    override fun getItemCount(): Int {
+        return mFilteredNotes.size
+    }
+
+    @Suppress("CyclomaticComplexMethod", "LongMethod")
+    override fun onBindViewHolder(noteViewHolder: NoteViewHolder, position: Int) {
+        val note = getNoteAtPosition(position) ?: return
+        noteViewHolder.mContentView.tag = note.id
+
+        // Display group header
+        val timeGroup = Note.getTimeGroupForTimestamp(note.timestamp)
+        var previousTimeGroup: NoteTimeGroup? = null
+        if (position > 0) {
+            val previousNote = getNoteAtPosition(position - 1)
+            previousTimeGroup = Note.getTimeGroupForTimestamp(
+                previousNote!!.timestamp
+            )
+        }
+        if (previousTimeGroup != null && previousTimeGroup == timeGroup) {
+            noteViewHolder.mHeaderText.visibility = View.GONE
+        } else {
+            noteViewHolder.mHeaderText.visibility = View.VISIBLE
+            if (timeGroup == NoteTimeGroup.GROUP_TODAY) {
+                noteViewHolder.mHeaderText.setText(R.string.stats_timeframe_today)
+            } else if (timeGroup == NoteTimeGroup.GROUP_YESTERDAY) {
+                noteViewHolder.mHeaderText.setText(R.string.stats_timeframe_yesterday)
+            } else if (timeGroup == NoteTimeGroup.GROUP_OLDER_TWO_DAYS) {
+                noteViewHolder.mHeaderText.setText(R.string.older_two_days)
+            } else if (timeGroup == NoteTimeGroup.GROUP_OLDER_WEEK) {
+                noteViewHolder.mHeaderText.setText(R.string.older_last_week)
+            } else {
+                noteViewHolder.mHeaderText.setText(R.string.older_month)
+            }
```

Comment:
These lines can be refactored as following code snippet:
```
 noteViewHolder.mHeaderText.setText(
    when(timeGroup) {
        NoteTimeGroup.GROUP_TODAY -> R.string.stats_timeframe_today
        ...etc
    }
)
```

Reply:
Resolved via [1a729ca](https://github.com/wordpress-mobile/WordPress-Android/pull/20129/commits/1a729ca46839c8ddba432221a50e2c9597f006c9)